Timeline.Core/BuiltInInterpolables.cs

```diff
@@ -276,16 +279,34 @@ private static void Animation()
                     {
                         OCIChar chara = (OCIChar)oci;
                         OICharInfo.AnimeInfo info = (OICharInfo.AnimeInfo)leftValue;
-                        if (chara.oiCharInfo.animeInfo.category != info.category || chara.oiCharInfo.animeInfo.group != info.group || chara.oiCharInfo.animeInfo.no != info.no)
+                        if (chara.oiCharInfo.animeInfo.group != info.group || chara.oiCharInfo.animeInfo.category != info.category || chara.oiCharInfo.animeInfo.no != info.no)
                             chara.LoadAnime(info.group, info.category, info.no);
                     },
                     interpolateAfter: null,
                     isCompatibleWithTarget: (oci) => oci is OCIChar,
                     getValue: (oci, parameter) =>
                     {
                         OICharInfo.AnimeInfo info = ((OCIChar)oci).oiCharInfo.animeInfo;
-                        return new OICharInfo.AnimeInfo() { category = info.category, group = info.group, no = info.no };
+                        return new OICharInfo.AnimeInfo() { group = info.group, category = info.category, no = info.no };
                     },
+#if KOIKATSU || SUNSHINE
+                    readValueFromXml: (parameter, node) =>
+                    {
+                        StudioResolveInfo resolveInfo = UniversalAutoResolver.LoadedStudioResolutionInfo.FirstOrDefault(x => x.Slot == node.ReadInt("valueNo") && x.GUID == node.Attributes?["GUID"]?.InnerText && x.Group == node.ReadInt("valueGroup") && x.Category == node.ReadInt("valueCategory"));
+                        return node.ReadInt("valueNo") >= UniversalAutoResolver.BaseSlotID && resolveInfo == null
+                            ? new OICharInfo.AnimeInfo() { group = 0, category = 0, no = 0 } // This prevents some of the potential LoadAnime spam
+                            : new OICharInfo.AnimeInfo() { group = node.ReadInt("valueGroup"), category = node.ReadInt("valueCategory"), no = resolveInfo != null ? resolveInfo.LocalSlot : node.ReadInt("valueNo") };
```

Comment:
`node.Read*` and `node.Attributes?["GUID"]?.InnerText` are expensive to run in a loop, the FirstOrDefault could take a long time to complete since there can be thousands of entries in the list. Instead read once into a variable at the start and reuse that.

Reply:
Thank you very much for your review!
But I'm not sure if this is what you asked for.
If this isn't the case, I need some more specific instructions or examples.